tasmota/tasmota_support/support.ino

```diff
@@ -590,6 +590,33 @@ char* Trim(char* p) {
   return p;
 }
 
+size_t HexStringToBinaryString(char** source) {
```

Comment:
If it's binary content, I would prefer `uint8_t **` instead of `char **`. The problem is that signness of `char` is architecture-dependant. So fixing to non-signed avoids some errors.

Reply:
According to the C and C++ standards, a byte has at least enough bits to store a character from the basic execution character set, which typically means at least 8 bits. Hence, in the context of standard conformant C and C++ compilers, char is always 1 byte.

Reply:
Oh, I didn't realize that you are changing the values in place. I'm not a big fan of a function that scraps the source string and converts to binary in place. Please have an input string (`const char *`) and an output buffer (`uint8_t *`) and a length of output buffer (`size_t`)

Reply:
Btw, if the caller wants to rewrite the input string, then it's their explicit choice when calling the function.

Reply:
Upon reviewing the support.ino file, I discovered a function: `int32_t HexToBytes(const char* hex, uint8_t* out, size_t* outLen)`, designed for converting hexadecimal to binary. However, this function exhibits some issues: firstly, outLen is initialised to zero and not subsequently updated. Secondly, there are no validations to confirm whether the input string is exclusively comprised of hexadecimal characters.

What course of action should I take in this situation?
1. Request a review of the function to solicit feedback or improvements from other team members.
2. Modify the function and include these changes in my existing pull request, which already contains the Publish3 command.
3. Make the necessary modifications to the function and initiate a separate pull request for these changes. Once this pull request is merged, proceed with another pull request for the Publish3 command.

Reply:
Thanks for your patience. You can proceed with 2. and include the changes in this PR